sycl/include/sycl/ext/intel/math.hpp

```diff
@@ -0,0 +1,69 @@
+//==------------- math.hpp - Intel specific math API -----------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// The main header of Intel specific math API
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#include <sycl/half_type.hpp>
+#include <type_traits>
+
+// _iml_half_internal is internal representation for fp16 type used in intel
+// math device library. The definition here should align with definition in
+// https://github.com/intel/llvm/blob/sycl/libdevice/imf_half.hpp
+#if defined(__SPIR__)
+typedef _Float16 _iml_half_internal;
+#else
+typedef uint16_t _iml_half_internal;
+#endif
+
+extern "C" {
+float __imf_saturatef(float);
+float __imf_copysignf(float, float);
+double __imf_copysign(double, double);
+_iml_half_internal __imf_copysignf16(_iml_half_internal, _iml_half_internal);
+};
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext {
+namespace intel {
+namespace math {
+
+#if __cplusplus >= 201703L
+template <typename Tp> Tp saturate(Tp x) {
+  static_assert(std::is_same<Tp, float>::value,
```

Comment:
```suggestion
  static_assert(std::is_same_v<Tp, float>,
```

We've checked for C++17 above, haven't we? Same below.